qualtran/bloqs/block_encoding/unitary.py

```diff
@@ -74,10 +75,7 @@ def pretty_name(self) -> str:
 
     @property
     def signal_state(self) -> BlackBoxPrepare:
-        # This method will be implemented in the future after PrepareOracle
-        # is updated for the BlockEncoding interface.
-        # Github issue: https://github.com/quantumlib/Qualtran/issues/1104
-        raise NotImplementedError
+        return BlackBoxPrepare(PrepareIdentity((QAny(self.ancilla_bitsize),)))
```

Comment:
Does this cause any issues when `ancilla_bitsize=0`?

Reply:
It would fail if we ever took that `signal_state` and did a decomposition. It's not something we do currently. 

I think a simple fix would be to add a check in `PrepareIdentity.build_composite_bloq` to not do anything if the bitsize is zero.

Reply:
If that (or some other change to `PrepareIdentity`) is desirable we can do in a follow-up PR.